Sources/VLCSettingsSpecifierManager.swift

```diff
@@ -0,0 +1,107 @@
+ * VLCSettingsSpecifierManager.swift
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import UIKit
+
+class VLCSettingsSpecifierManager: NSObject {
+    var specifier: IASKSpecifier!
+    var settingsReader: IASKSettingsReader!
+    var settingsStore: IASKSettingsStore!
+    
+    var items: NSArray {
+        guard let items = specifier.multipleValues() as NSArray? else {
+            fatalError("VLCSettingsSpecifierManager: No rows provided for \(specifier.key())")
+        }
+        return items
+    }
+    
+    var selectedIndex: IndexPath {
+        var index: Int
+        if let selectedItem = settingsStore.object(forKey: specifier.key()) {
+            index = items.index(of: selectedItem)
+        } else {
+            index = items.index(of: specifier.defaultValue())
+        }
+        return IndexPath(row: index, section: 0)
+    }
+    
+    @objc init(specifier: IASKSpecifier, settingsReader: IASKSettingsReader, settingsStore: IASKSettingsStore) {
+        self.specifier = specifier
+        self.settingsReader = settingsReader
+        self.settingsStore = settingsStore
+    }
+    
+    fileprivate func updateCollectionViewCellApparence(cell: VLCSettingsSheetCell, highlighted: Bool) {
+        var textColor = PresentationTheme.current.colors.cellTextColor
+        var tintColor = PresentationTheme.current.colors.cellDetailTextColor
+        
+        if highlighted {
+            textColor = PresentationTheme.current.colors.orangeUI
+            tintColor = PresentationTheme.current.colors.orangeUI
+        }
+        
+        cell.tintColor = tintColor
+        cell.name.textColor = textColor
+        cell.checkmark.isHidden = !highlighted
+    }
+}
+
+// MARK: VLCActionSheetDelegate
+
+extension VLCSettingsSpecifierManager: VLCActionSheetDelegate {
+    func headerViewTitle() -> String? {
+        return specifier.title()
+    }
+    
+    func itemAtIndexPath(_ indexPath: IndexPath) -> Any? {
+        return items[indexPath.row]
+    }
+    
+    func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath) {
+        if indexPath == selectedIndex {
+            return
+        }
+
+        guard let oldCell = collectionView.cellForItem(at: selectedIndex) as? VLCSettingsSheetCell,
+            let newCell = collectionView.cellForItem(at: indexPath) as? VLCSettingsSheetCell else {
+                assertionFailure("VLCSettingsSpecifierManager: VLCActionSheetDelegate: Cell is not a VLCSettingsSheetCell")
+                return
+        }
+
+        updateCollectionViewCellApparence(cell: oldCell, highlighted: false)
```

Comment:
instead of highlighted does the cell maybe know that it's highlighted or selected itself? Maybe you can overwrite the setter if that's the case